tests/util/test_logging_filter.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import annotations
+
+import logging
+from re import split
+from time import sleep
+
+import pytest
+
+from chia.util.logging import TimedDuplicateFilter
+
+
+def test_logging_filter(caplog: pytest.LogCaptureFixture) -> None:
+
+    log_interval_secs = 10
+    num_logs = 11
+    sleep_secs = 1
+
+    log = logging.getLogger()
+    log.addFilter(TimedDuplicateFilter("Filter this log message.*", log_interval_secs))
+
+    for n in range(num_logs):
+        with caplog.at_level(logging.WARNING):
+            log.warning(f"Filter this log message {n}")
+            sleep(sleep_secs)
+
+    assert len(split("\n", caplog.text)) <= ((num_logs * sleep_secs) / log_interval_secs) + 1
```

Comment:
I think this test is flaky. each iteration of the loop will add rounding error to the total time. you may want to check how much time has passed each loop iteration as well, and just sleep long enough to get to the next second. It would still be flaky because there are no guarantees, but probably less so

Reply:
Thanks, done.